modules/src/ics02_client/context_mock.rs

```diff
@@ -5,73 +5,73 @@ use crate::ics02_client::error::Error;
 use crate::ics24_host::identifier::ClientId;
 use crate::mock_client::header::MockHeader;
 use crate::mock_client::state::{MockClientState, MockConsensusState};
+use std::collections::HashMap;
 use tendermint::block::Height;
 
+/// A mock implementation of client context. Implements traits `ClientReader`
+/// and `ClientKeeper`, and is therefore useful for testing the ICS02 handlers (create, update
+/// client) and other dependent ICS handlers (e.g., ICS03).
 #[derive(Clone, Debug, PartialEq)]
 pub struct MockClientContext {
-    pub client_id: ClientId,
-    pub client_state: Option<MockClientState>,
+    /// The set of all clients, as well as their states, which this context comprises.
+    pub client_states: HashMap<ClientId, MockClientState>,
+    /// All clients in this context have this (same) type.
```

Comment:
We should build the mock context to allow multiple clients with different types. It should be a collection of items very similar to what used to be `MockClientContext` (e.g. a `HashMap` with key `ClientId` and value that includes type, state and a collection of consensus states).